jscomp/others/js_re.mli

```diff
@@ -121,9 +125,10 @@ while not !break do
   match re |> Js.Re.exec str with
   | None -> break := true
   | Some result ->
-    let match_ = (Js.Re.matches result).(0) in
-    let next = string_of_int (Js.Re.lastIndex re) in
-    Js.log ("Found " ^ match_ ^ ". Next match starts at " ^ next)
+    let _ = Js.Nullable.bind (Js.Re.captures result).(0) ((fun match_ ->
```

Comment:
Since the value is just thrown away anyway, we could just use `Js.Nullable.iter` here instead, and thereby avoiding explicitly return `()` as well.

Reply:
Also, in general, it's a bad idea to use just `let _ = ...` since you throw away type-checking then. Might not seem like a huge issue, but you'd be surprised how often you'll accidentally partially apply a function and not notice it because you just throw away the value (and type). Unfortunately you also throw away the side-effects in that case, which is a tricky bug top catch. But if you add a type annotation, you'll guard against that situation while still throwing away the value: `let _ : unit Js.nullable = ...`